src/windows/wsladiag/main.cpp

```diff
@@ -0,0 +1,44 @@
+#include <iostream>
+#include <windows.h>
+
+int wmain(int argc, wchar_t** argv)
+{
+    bool list = false;
```

Comment:
Since this is a new main() function, we need to do all the initialization like we do for processes. For now we probably to add: 

```
wsl::windows::common::wslutil::ConfigureCrt();
wsl::windows::common::wslutil::InitializeWil();

WslTraceLoggingInitialize(LxssTelemetryProvider, !wsl::shared::OfficialBuild);
auto cleanupTelemetry = wil::scope_exit_log(WI_DIAGNOSTICS_INFO, []() { WslTraceLoggingUninitialize(); });


wsl::windows::common::wslutil::SetCrtEncoding(_O_U8TEXT);

auto coInit = wil::CoInitializeEx(COINIT_MULTITHREADED);
wsl::windows::common::wslutil::CoInitializeSecurity();

WSADATA data;
THROW_IF_WIN32_ERROR(WSAStartup(MAKEWORD(2, 2), &data));

```

A few of those will be required before we can actually call into the COM API